packages/cli/src/util/get-deployment.ts

```diff
@@ -27,12 +29,17 @@ export default async function getDeployment(
   }
 
   try {
-    return await client.fetch<Deployment>(
+    return client.fetch<Deployment>(
       `/v13/deployments/${encodeURIComponent(hostOrId)}`
     );
   } catch (err: unknown) {
     if (isAPIError(err)) {
       if (err.status === 404) {
+        if (/^[A-Za-z0-9]+$/.test(hostOrId)) {
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏use strict check

This check makes me nervous that it will hit something we didn't intend. Can we check for the value starting with `dpl_` explicitly?

Reply:
`hostOrId` is either a URL or a deployment id. Both a URL and a valid deployment id will fail that regex because they contain `.:/_-`. If a `hostOrId` is only alpha-numeric, then the code simply retries with the `dpl_` prefix. We specifically do not want this conditional to check for `dpl_`.

Reply:
Is this because we strip `dpl_` in the deployment inspector URL? So the thinking is that a user could have copy/pasted that value to this command?

Reply:
@TooTallNate yes. Good idea? Bad idea?

Reply:
I'm undecided 😅 But I do think that if this were to be supported it should be done at the API level.

Reply:
OK, reverted auto `dpl_` prefixing.